src/server/internalServer.cpp

```diff
@@ -775,7 +777,14 @@ std::unique_ptr<Response> InternalServer::handle_content(const RequestContext& r
 
   const std::shared_ptr<Reader> reader = get_reader(bookName);
   if (reader == nullptr) {
-    return Response::build_404(*this, request, bookName);
+    std::string searchURL = m_root+"/search?pattern="+pattern;
+    MustacheData data;
+    data.set("pattern", pattern);
+    data.set("searchURL", searchURL);
+    kainjow::mustache::mustache tmpl("Make a full text search for <a href=\"{{searchURL}}\">{{{pattern}}}</a>");
```

Comment:
The other way around - `searchURL` must be substituted as is, and `pattern` must be HTML encoded.